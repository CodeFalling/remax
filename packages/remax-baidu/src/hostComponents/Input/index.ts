import * as React from 'react';
import { createHostComponent } from '@remax/shared';
import { BaseProps } from '../../types/component';

export interface InputProps extends BaseProps {
  value?: string;
  name?: string;
  type?: 'text' | 'number' | 'idcard' | 'digit';
  password?: boolean;
  placeholder?: string;
  placeholderStyle?: React.CSSProperties;
  placeholderClassName?: string;
  disabled?: boolean;
  maxLength?: number;
  cursorSpacing?: number;
  focus?: boolean;
  confirmType?: 'send' | 'search' | 'next' | 'go' | 'done';
  confirmHold?: boolean;
  cursor?: number;
  selectionStart?: number;
  selectionEnd?: number;
  adjustPosition?: boolean;
  onInput?: (event: any) => void;
  onFocus?: (event: any) => void;
  onBlur?: (event: any) => void;
  onConfirm?: (event: any) => void;
}

export const Input: React.ComponentType<InputProps> = createHostComponent<InputProps>('input');

Input.defaultProps = {
  type: 'text',
  password: false,
  placeholderClassName: 'input-placeholder',
  disabled: false,
  maxLength: 140,
  cursorSpacing: 0,
  focus: false,
  confirmHold: false,
  selectionStart: -1,
  selectionEnd: -1,
  adjustPosition: true,
};
