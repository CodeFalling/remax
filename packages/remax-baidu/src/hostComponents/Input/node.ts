export const alias = {
  id: 'id',
  className: 'class',
  style: 'style',
  value: 'value',
  type: 'type',
  password: 'password',
  placeholder: 'placeholder',
  placeholderStyle: 'placeholder-style',
  placeholderClassName: 'placeholder-class',
  disabled: 'disabled',
  maxLength: 'maxlength',
  cursorSpacing: 'cursor-spacing',
  focus: 'focus',
  confirmType: 'confirm-type',
  confirmHold: 'confirm-hold',
  cursor: 'cursor',
  selectionStart: 'selection-start',
  selectionEnd: 'selection-end',
  adjustPosition: 'adjust-position',
  onInput: 'bindinput',
  onFocus: 'bindfocus',
  onBlur: 'bindblur',
  onConfirm: 'bindconfirm',
};

export const props = Object.values(alias);
