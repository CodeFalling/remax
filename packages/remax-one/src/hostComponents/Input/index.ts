import * as React from 'react';
import { InputEvent } from '../../types';
import { createCallback, createInputEvent } from '../../createHostComponent';

export interface InputProps {
  // 通用属性
  readonly dataset?: DOMStringMap;
  id?: string;
  className?: string;
  style?: React.CSSProperties;
  /** 输入框的初始内容 */
  defaultValue?: string;
  value?: string;
  name?: string;
  /** input 的类型 */
  type?: string;
  /** 是否是密码类型 */
  password?: boolean;
  /** 输入框为空时占位符 */
  placeholder?: string;
  placeholderStyle?: React.CSSProperties;
  /** 是否禁用 */
  disabled?: boolean;
  /** 最大输入长度，设置为 -1 的时候不限制最大长度 */
  maxlength?: number;
  /** 获取焦点 */
  focus?: boolean;
  onInput?: (e: InputEvent) => any;
  onConfirm?: (e: InputEvent) => void;
  onFocus?: (e: InputEvent) => void;
  onBlur?: (e: InputEvent) => void;
}

export interface InputState {
  value?: string;
  controlled: boolean;
}

export default class Input extends React.Component<InputProps, InputState> {
  static defaultProps = {
    'toutiao-maxlength': 140,
    'toutiao-selection-end': 999,
    'toutiao-selection-start': 999,
    'wechat-maxlength': 140,
    'wechat-selection-end': 999,
    'wechat-selection-start': 999,
  };

  state: InputState = {
    value: '',
    // 支付宝
    controlled: false,
  };

  constructor(props: InputProps) {
    super(props);

    const controlled = props.value !== undefined;
    const value = controlled ? props.value : props.defaultValue;

    this.state = {
      value,
      controlled,
    };
  }

  componentDidUpdate() {
    if (this.props.value !== undefined && this.props.value !== this.state.value) {
      this.setState({ value: this.props.value });
    }
  }

  handleInput = (e: InputEvent) => {
    const { controlled } = this.state;

    if (!controlled) {
      this.setState({ value: e.target.value });
    }

    if (typeof this.props.onInput === 'function') {
      return this.props.onInput(e);
    }

    // 微信
    // 注意，微信要对 input 受控，必须要在 onInput 方法返回值
    return controlled ? this.props.value : e.target.value;
  };

  render() {
    const inputProps = {
      ...this.props,
      onInput: createCallback(this.handleInput, createInputEvent),
    };

    if (inputProps.onConfirm) {
      inputProps.onConfirm = createCallback(this.props.onConfirm, createInputEvent);
    }

    if (inputProps.onFocus) {
      inputProps.onFocus = createCallback(this.props.onFocus, createInputEvent);
    }

    if (inputProps.onBlur) {
      inputProps.onBlur = createCallback(this.props.onBlur, createInputEvent);
    }

    return React.createElement('input', {
      ...inputProps,
      ...this.state,
    });
  }
}
