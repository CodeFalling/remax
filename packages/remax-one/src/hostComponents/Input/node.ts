const platformAlias: any = {
  alipay: {
    onInput: 'onInput',
    onConfirm: 'onConfirm',
    onFocus: 'onFocus',
    onBlur: 'onBlur',
    controlled: 'controlled',
  },
  wechat: {
    onInput: 'bindinput',
    onConfirm: 'bindconfirm',
    onFocus: 'bindfocus',
    onBlur: 'bindblur',
  },
  toutiao: {
    onInput: 'bindinput',
    onConfirm: 'bindconfirm',
    onFocus: 'bindfocus',
    onBlur: 'bindblur',
  },
};

const defaultAlias = {
  id: 'id',
  className: 'class',
  style: 'style',
  value: 'value',
  defaultValue: 'value',
  name: 'name',
  type: 'type',
  password: 'password',
  placeholder: 'placeholder',
  disabled: 'disabled',
  maxlength: 'maxlength',
  focus: 'focus',
  confirmType: 'confirm-type',
  confirmHold: 'confirm-hold',
  placeholderStyle: 'placeholder-style',
};

export const alias = { ...defaultAlias, ...platformAlias[process.env.REMAX_PLATFORM!] };

export const props = Object.values(alias);
