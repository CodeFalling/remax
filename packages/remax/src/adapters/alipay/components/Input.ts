import factory from './factory';

export interface InputProps {
  id?: string;
  value?: string;
  name?: string;
  type?: string;
  password?: boolean;
  placeholder?: string;
  placeholderStyle?: string;
  placeholderClass?: string;
  disabled?: boolean;
  maxlength?: number;
  focus?: boolean;
  confirmType?: string;
  confirmHold?: boolean;
  cursor?: number;
  selectionStart?: number;
  selectionEnd?: number;
  randomNumber?: boolean;
  controlled?: boolean;
  onInput?: (e: any) => void;
  onConfirm?: (e: any) => void;
  onFocus?: (e: any) => void;
  onBlur?: (e: any) => void;
}
const Input = factory<InputProps>('input');

export default Input;
