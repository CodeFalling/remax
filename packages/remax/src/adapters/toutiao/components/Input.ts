import { CSSProperties } from 'react';
import factory from './factory';

export interface InputProps {
  focus?: boolean;
  maxlength?: number;
  value: string;
  password?: boolean;
  type: 'text' | 'number' | 'digit';
  cursor?: number;
  selectionStart?: number;
  selectionEnd?: number;
  adjustPosition?: boolean;
  placeholder?: string;
  placeholderStyle?: CSSProperties;
  placeholderClass?: string;
  disabled?: boolean;
  cursorSpacing?: number;
  onInput?: (e: any) => void | string;
  onClick?: (e: any) => void;
  onFocus?: (e: any) => void;
  onBlur?: (e: any) => void;
  onConfirm?: () => void;
}

export default factory<InputProps>('input');
