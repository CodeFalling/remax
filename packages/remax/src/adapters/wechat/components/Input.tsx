import React, { CSSProperties, FunctionComponent, useState } from 'react';
import propsAlias from './propsAlias';

export interface InputProps {
  id?: string;
  autoFocus?: boolean;
  className?: string;
  focus?: boolean;
  name?: string;
  maxlength?: number;
  value: any;
  style?: CSSProperties;
  password?: string;
  type: 'text' | 'number' | 'idcard' | 'digit';
  confirmType: 'send' | 'search' | 'next' | 'go' | 'done';
  confirmHold?: boolean;
  cursor?: number;
  selectionStart?: number;
  selectionEnd?: number;
  adjustPosition?: boolean;
  placeholder?: string;
  placeholderStyle?: CSSProperties;
  placeholderClass?: string;
  disabled?: boolean;
  cursorSpacing?: number;
  onInput?: (...params: any) => void;
  onClick?: (...params: any) => void;
  onFocus?: (...params: any) => void;
  onBlur?: () => void;
  onConfirm?: () => void;
  onKeyboardHeightChange?: () => void;
}

interface State {
  focus: boolean;
}

function useInnerFocus(
  initialValue?: boolean
): [boolean, typeof handleInnerFocus] {
  const [innerFocus = false, setInnerFocus] = useState(initialValue);

  const handleInnerFocus = (func?: Function) => (...params: any) => {
    if (!innerFocus) {
      setInnerFocus(true);
    }

    if (typeof func === 'function') {
      return func(...params);
    }
  };

  return [innerFocus, handleInnerFocus];
}

const Input: FunctionComponent<InputProps> = props => {
  const {
    autoFocus,
    children,
    focus,
    onInput,
    onClick,
    onFocus,
    ...restProps
  } = props;
  const [innerFocus, handleInnerFocus] = useInnerFocus(focus || autoFocus);

  const inputProps = propsAlias({
    ...restProps,
    autoFocus,
    focus: innerFocus,
    onInput: handleInnerFocus(onInput),
    onClick: handleInnerFocus(onClick),
    onFocus: handleInnerFocus(onFocus),
  });

  return React.createElement('input', inputProps, children);
};

Input.defaultProps = {
  maxlength: -1,
};

export default Input;
