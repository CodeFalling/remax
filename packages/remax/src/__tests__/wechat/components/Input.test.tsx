import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Input from '../../../adapters/wechat/components/Input';

describe('Input', () => {
  it('render correctly', () => {
    const testRenderer = TestRenderer.create(<Input value="" type="text" />);

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
