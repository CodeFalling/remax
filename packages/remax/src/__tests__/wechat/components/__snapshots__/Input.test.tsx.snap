// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render correctly 1`] = `
<input
  bindblur={[Function]}
  bindfocus={[Function]}
  bindinput={[Function]}
  bindtap={[Function]}
  focus={false}
  maxlength={140}
  selectionEnd={999}
  selectionStart={999}
  type="text"
  value=""
/>
`;
